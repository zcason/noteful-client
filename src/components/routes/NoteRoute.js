import React, { Component } from 'react';
import MainSection from '../sections/MainSection';
import SidebarSection from '../sections/SidebarSection';
import { withRouter } from 'react-router-dom';
import ErrorBoundary from '../../errors/ErrorBoundary';
import PropTypes from 'prop-types';

class NoteRoute extends Component {
    render(){
        const store = {
            notes: this.props.store.notes.filter( note => this.props.history.location.pathname.includes(note.id) ),
            folders: this.props.store.folders
        }
        return (
            <div className="App">
                <ErrorBoundary message="Sidebar Section Error">
                    <SidebarSection 
                        store={store}
                    />
                </ErrorBoundary>
                <ErrorBoundary message="Main Section Error">
                    <MainSection 
                        store={store}
                    />
                </ErrorBoundary>

            </div>
       );
    }
}

NoteRoute.propTypes = {
    store: PropTypes.object.isRequired
}

export default withRouter(NoteRoute);