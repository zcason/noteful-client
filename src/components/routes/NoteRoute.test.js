import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import store from '../../dummy-store';
import NoteRoute from './NoteRoute';

describe(`NoteRoute component`, () => {
    const props = {
      store: store
    }
  
    it('renders NoteRoute by default', () => {
      const wrapper = shallow(<NoteRoute />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  
    it('renders note route from store when in props', () => {
      const component = shallow(<NoteRoute {...props} />)
        .find('.App')
      expect(toJson(component)).toMatchSnapshot()
    })
  })
  