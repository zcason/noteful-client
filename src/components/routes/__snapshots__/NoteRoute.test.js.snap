// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteRoute component renders NoteRoute by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`NoteRoute component renders note route from store when in props 1`] = `null`;
